script/functions.sh

```diff
@@ -11,6 +11,14 @@ export JRUBY_OPTS=${JRUBY_OPTS:-"--server -Xcompile.invokedynamic=false"}
 SPECS_HAVE_RUN_FILE=specs.out
 MAINTENANCE_BRANCH=`cat maintenance-branch`
 
+# Don't allow rubygems to pollute what's loaded. Also, things boot faster
+# without the extra load time of rubygems. Only works on MRI Ruby 1.9+
+if is_mri_192_plus; then
+  export RUBYOPT="--disable=gem"
+else
+  export RUBYOPT=""
```

Comment:
I believe this line might be causing the JRuby failures -- perhaps on JRuby the `RUBYOPT` env var already has a value?  Here you are clearing it.  Is this line necessary?